src/sampler.jl

```diff
@@ -1,3 +1,55 @@
+##
+## Interface functions
+##
+
+function init(rng::AbstractRNG, h::Hamiltonian, θ::AbstractVector{T}) where {T<:Real}
+    # Ensure h.metric has the same dim as θ.
+    h = update(h, θ)
+    # Prepare phase point for sampling
+    r = rand(rng, h.metric)
+    z = phasepoint(h, θ, r)
+    return h, z
+end
+
+function step(rng::AbstractRNG, h::Hamiltonian, τ::AbstractProposal, z::PhasePoint)
```

Comment:
Perhaps try to match the function signature with the following if possible?

https://github.com/TuringLang/Turing.jl/blob/csp/interface-pt2/src/interface/Interface.jl#L238

We can probably treat `h::Hamiltonian` as an equivalent to `ModelType`.

Reply:
What do you mean by the signature here?

Reply:
I mean function arguments.

Reply:
You mean the `N` and `debug` in the function? 

```julia
function step!(
    rng::AbstractRNG,
    ℓ::ModelType,
    s::SamplerType,
    N::Integer;
    debug::Bool=false,
    kwargs...
) where {ModelType<:Sampleable, SamplerType<:AbstractSampler}
    # Do nothing.
    debug && @warn "No step! function has been implemented for objects of types \n- $(typeof(ℓ)) \n- $(typeof(s))"
end
```

I don't see other parts could be mimiced.

Reply:
I see. It makes sense.

Reply:
So here sampler is actually closer to `τ::AbstractProposal`.